plugins/services/src/js/events/MarathonActions.js

```diff
@@ -50,6 +58,61 @@ function buildURI(path) {
   return `${Config.rootUrl}${Config.marathonAPIPrefix}${path}`;
 }
 
+const deleteGroupRequest = function(url) {
+  RequestUtil.json({
+    url,
+    method: "DELETE",
+    success() {
+      AppDispatcher.handleServerAction({
+        type: REQUEST_MARATHON_GROUP_DELETE_SUCCESS
+      });
+    },
+    error(xhr) {
+      AppDispatcher.handleServerAction({
+        type: REQUEST_MARATHON_GROUP_DELETE_ERROR,
+        data: RequestUtil.parseResponseBody(xhr),
+        xhr
+      });
+    }
+  });
+};
+
+const handleSDKDelete = function(
+  payload,
+  frameworkId,
+  dispatcher,
+  resolve,
+  reject
+) {
+  const { action, xhr } = payload;
+  const eventIsForThisFramework = action && action.appId
+    ? action.appId === frameworkId
+    : false;
+  if (!eventIsForThisFramework) return;
+  if (action.type === REQUEST_COSMOS_PACKAGE_UNINSTALL_SUCCESS) {
+    AppDispatcher.unregister(dispatcher);
+    resolve();
+  } else if (action.type === REQUEST_COSMOS_PACKAGE_UNINSTALL_ERROR) {
+    AppDispatcher.unregister(dispatcher);
+    if (xhr && xhr.response) {
+      var errorMessage = JSON.parse(xhr.response);
+      if (errorMessage && errorMessage.message) {
+        reject(errorMessage);
```

Comment:
It's probably a good idea to `return` here to prevent continued execution of this function.